### PR TITLE
Fix bug 1528177 - Add links to Sign in and out in Translate.Next.

### PR DESCRIPTION
**Pull request by adngdb**
This also does a bunch of refactoring, putting all user-related components into the user module and exposing a single, catch-all UserControls component for simplicity of use.

### CODE REVIEW

**Comment by mathjazz** [comment]
Nice work! A couple of notes:

1. After successful login, you get redirected to the homepage. Instead, you should end up on the same page, like in Translate.Current:
https://mozilla-pontoon-staging.herokuapp.com/sl/firefox/all-resources/?string=74127

2. There should be no confirmation step after Signing Out. Not only because such UX is inconsistent with the Translate.Current UX, which we're copying, but also because signing in and out isn't an irreversible step. Additionally, we'll change the Sign in/Sign out UX soon [anyways](https://bugzilla.mozilla.org/show_bug.cgi?id=1486506).

File: frontend/src/core/user/components/SignInLink.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        const base = this.props.url;
+        const startSign = base.match(/\?/) ? '&': '?';
+
+        return (
+            base +
+            startSign +
+            'next=' +
+            window.location.pathname +
+            window.location.search
+        );
```
Comment (mathjazz) [inline frontend/src/core/user/components/SignInLink.js]: Is there a more elegant way of appending a parameter to the URL?

File: frontend/src/core/user/components/UserControls.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    render() {
+        const { router, user } = this.props;
+
+        return <div className='user-controls'>
+            <UserAutoUpdater getUserData={ this.getUserData } />
+
+            { user.isAuthenticated ?
+                <SignOut dispatch={ this.props.dispatch } url={ user.signOutURL } /> :
```
Comment (mathjazz) [inline frontend/src/core/user/components/UserControls.js]: I haven't seen `dispatch` being passed around like this. Am I doing something wrong?

**Comment by mathjazz** [comment]
@adngdb This now works as expected from my perspective. Could you please r? (I cannot assign a review to you, because it's your branch).

File: frontend/src/core/user/components/UserControls.js (hunk 2 above)
Comment (adngdb) [reply frontend/src/core/user/components/UserControls.js]: Yes, `dispatch` should never be passed around. Instead you should create a new method here that uses `dispatch` and pass that method down. I can take it up from here though, let me review by pushing some commits. :sweat_smile:

**Comment by adngdb** [comment]
@mathjazz I think this is in working order. Wanna take a last look at it and merge if you're happy with it?

**Pull request merged**